src/Datadog.Trace/ExtensionMethods/SpanExtensions.cs

```diff
@@ -67,6 +71,26 @@ public static void AddTagsFromDbCommand(this Span span, IDbCommand command)
             }
         }
 
+        internal static void ApplyHeaderTags(this Span span, IHeadersCollection headers, IDictionary<string, string> headerTags)
+        {
+            if (headerTags is not null && !headerTags.IsEmpty())
+            {
+                try
+                {
+                    // extract propagation details from http headers
+                    var tagsFromHeaders = SpanContextPropagator.Instance.ExtractHeaderTags(headers, headerTags);
+                    foreach (KeyValuePair<string, string> kvp in tagsFromHeaders)
+                    {
+                        span.SetTag(kvp.Key, kvp.Value);
```

Comment:
According to the docs (which we failed to follow when we originally added this feature), tag names should be:
```
http.<request_or_response>.headers.<normalized_header_name>
```
Also, we already use `http.request.headers.host`. Should we change this to follow the RFC? And if we do, should we keep using the current tag name for requests in addition to the new name (for backwards compat)?